### PR TITLE
Add test_scheduling skipped cases

### PR DESCRIPTION
**Pull request by chriscchien**
Add test_scheduling skipped cases
- test_allow_empty_node_selector_volume_setting
- test_global_disk_soft_anti_affinity
- test_allow_empty_disk_selector_volume_setting
- test_volume_disk_soft_anti_affinity

ref: [7339](https://github.com/longhorn/longhorn/issues/7339)

### CODE REVIEW

**Comment by chriscchien** [comment]
Test result of test_allow_empty_node_selector_volume_setting
image

File: manager/integration/tests/test_scheduling.py [hunk 1]
```diff
@@ -1754,7 +1758,37 @@ def test_allow_empty_node_selector_volume_setting(): # NOQA
...
+    volume = wait_for_volume_detached(client, volume_name)
+
+    volume = client.by_id_volume(volume.name)
+    volume = wait_for_volume_condition_scheduled(client, volume_name,
+                                                 "status",
+                                                 CONDITION_STATUS_FALSE)
+
+    # Rremove tag from 1 node and set setting allow-empty-node-selector-volume
+    # to true
+    node = client.by_id_node(get_self_host_id())
```
Comment (yangchiu) [inline manager/integration/tests/test_scheduling.py:1780]: @chriscchien Do you know why we have to remove tag from a node?

cc @ChanYiLin

Comment (chriscchien) [reply manager/integration/tests/test_scheduling.py:1780]: To verify that when `allow-empty-node-selector-volume = true`, no matter node have tags or not,  all replicas can be scheduled on those nodes.

cc @ChanYiLin

**Comment by chriscchien** [comment]
Test [result](https://ci.longhorn.io/job/private/job/longhorn-tests-regression/5407/testReport/tests/test_scheduling/) of `test_global_disk_soft_anti_affinity`

**Comment by chriscchien** [comment]
Add test case `test_allow_empty_disk_selector_volume_setting` and `test_volume_disk_soft_anti_affinity`

Test result for those cases
- [test_allow_empty_node_selector_volume_setting](https://ci.longhorn.io/job/private/job/longhorn-tests-regression/5433/testReport/tests/test_scheduling/)
- [test_global_disk_soft_anti_affinity](https://ci.longhorn.io/job/private/job/longhorn-tests-regression/5428/testReport/tests/test_scheduling/)
- [test_allow_empty_disk_selector_volume_setting](https://ci.longhorn.io/job/private/job/longhorn-tests-regression/5429/testReport/tests/test_scheduling/)
- [test_volume_disk_soft_anti_affinity](https://ci.longhorn.io/job/private/job/longhorn-tests-regression/5430/testReport/tests/test_scheduling/)

**Comment by innobead** [comment]
Because you have 4 cases here, it's better to have 4 commits for each for better commit history. Do this way next time.

**Comment by chriscchien** [comment]
> This pull request is now in conflicts. Could you fix it @chriscchien? 🙏

Updated

**Review by yangchiu — APPROVED** [review]
LGTM

**Comment by chriscchien** [comment]
@mergify backport v1.6.x

**Pull request merged**